crates/pipeline-manager/src/api/endpoints/pipeline_interaction/lifecycle_events.rs

```diff
@@ -0,0 +1,80 @@
+use actix_web::{
+    get,
+    web::{self, Data as WebData, ReqData},
+    HttpResponse,
+};
+
+use crate::{
+    api::{examples, main::ServerState},
+    db::{storage::Storage, types::tenant::TenantId},
+    error::ManagerError,
+};
+
+/// Query parameters for lifecycle events endpoint.
+#[derive(serde::Deserialize)]
+pub struct EventsParameters {
+    pub max_events: u32,
+}
+
+/// Get lifecycle events for a pipeline.
+///
+/// Returns a list of lifecycle events for the specified pipeline.
+/// The number of events returned is controlled by the `max_events` query parameter.
+///
+/// # Parameters
+/// - `pipeline_name`: Unique pipeline name (path parameter)
+/// - `max_events`: Maximum number of events to return (query parameter, required)
+///
+/// # Returns
+/// - 200 OK: List of lifecycle events for the pipeline
+/// - 404 NOT_FOUND: Pipeline with that name does not exist
+/// - 503 SERVICE_UNAVAILABLE: Disconnected or timeout during response
```

Comment:
I don't think this specific 503 can happen, nothing is being HTTP interacted with?